File: 정렬/프로그래머스Lv0-특이한 정렬.js
function solution(numlist, n) {
  return numlist.sort((a, b) => {
    // 만약 절대값이 같다면 더 큰 값을 앞에다 두기
    if (Math.abs(a - n) === Math.abs(b - n)) {
      return a < b ? 1 : -1
    }
    // 절대값 차이를 기준으로 오름차순 정렬
    return Math.abs(a - n) - Math.abs(b - n)
  })
}
// 정확성  테스트
// 테스트 1 〉	통과 (0.15ms, 33.4MB)
// 테스트 2 〉	통과 (0.17ms, 33.5MB)
// 테스트 3 〉	통과 (0.14ms, 33.4MB)
// 테스트 4 〉	통과 (0.15ms, 33.5MB)
// 테스트 5 〉	통과 (0.06ms, 33.4MB)
// 테스트 6 〉	통과 (0.13ms, 33.5MB)
// 테스트 7 〉	통과 (0.03ms, 33.6MB)

// 이런것도 가능;;
function solution(numlist, n) {
  return numlist.sort((a, b) => Math.abs(a - n) - Math.abs(b - n) || b - a)
}

File: 힙(우선순위 큐)/프로그래머스Lv0-특이한 정렬.py
import heapq
def find_standard_index(numlist, n):
    index = 0
    for i, number in enumerate(numlist):
        if abs(number - n) < abs(numlist[index] - n):
            index = i
    return index

def solution(numlist, n):
    heap = []
    index = find_standard_index(numlist, n)
    
    for i, number in enumerate(numlist):
        heapq.heappush(heap, (abs(number - n), -number))
    
    result = []
    while heap:
        diff, opposite_number = heapq.heappop(heap)
        result.append(-opposite_number)
    return result
"""
정확성  테스트
테스트 1 〉	통과 (0.05ms, 10.2MB)
테스트 2 〉	통과 (0.07ms, 10.3MB)
테스트 3 〉	통과 (0.05ms, 10.3MB)
테스트 4 〉	통과 (0.04ms, 10.2MB)
테스트 5 〉	통과 (0.01ms, 10.2MB)
테스트 6 〉	통과 (0.02ms, 10.2MB)
테스트 7 〉	통과 (0.00ms, 10.4MB)
"""

# 근데 걍 정렬로 풀면 됨;;
def solution(numlist, n):
    return sorted(numlist, key = lambda x: (abs(x - n), -x))
"""
정확성  테스트
테스트 1 〉	통과 (0.03ms, 10.3MB)
테스트 2 〉	통과 (0.05ms, 10.2MB)
테스트 3 〉	통과 (0.02ms, 10.3MB)
테스트 4 〉	통과 (0.03ms, 10.2MB)
테스트 5 〉	통과 (0.01ms, 10.1MB)
테스트 6 〉	통과 (0.01ms, 10.2MB)
테스트 7 〉	통과 (0.00ms, 10.2MB)
"""